### PR TITLE
Upgrade tools

### PR DESCRIPTION
**Pull request by chrlwrd**
Changes:
- Added GitHub actions to lint Dockerfile and bash scripts
- Added fluxctl + stern
- Upgrade various tools (including Terraform and Terragrunt in preparation for TF 0.13 soon)
- Shrink image size slightly, remove layers
- Clean up linting errors from Dockerfile and bash scripts

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -69,32 +70,32 @@ ENV DEBIAN_FRONTEND=noninteractive \
...
+    find /usr/share/doc -depth -type f ! -name copyright -print0 | xargs -0 rm || true && \
+    find /usr/share/doc -empty -print0 | xargs -0 rmdir || true && \
     rm -rf /usr/share/man/* /usr/share/groff/* /usr/share/info/* && \
     rm -rf /usr/share/lintian/* /usr/share/linda/* /var/cache/man/*
 
-WORKDIR /tmp
-
 RUN apt-get -y update && \
-    apt-get -y install locales && \
+    apt-get -y install locales --no-install-recommends && \
```
Comment (primeroz) [inline Dockerfile:89]: can we add an editor to the image ?

like a vim-tiny ?

File: resources/libs/vault.sh [hunk 2]
```diff
@@ -10,33 +12,32 @@ GITOPS_CI_CONTAINER_VERSION="${GITOPS_CI_CONTAINER_VERSION:-0.9.0}"
...
+    mkdir -p "$CONFS_DIR/$env/kustomize"
+    mkdir -p "$CONFS_DIR/$env/yamls"
+    kustomize build "$(dirname "$k")" >"$CONFS_DIR/$env/kustomize/manifests.yaml"
 
     file=$CONFS_DIR/$env/kustomize/manifests.yaml
-    N_DOCS=$(cat $file | egrep ^kind | wc -l)
+    N_DOCS=$(grep -E ^kind "$file" -c)
 
-    let N_DOCS-=1
+    ((_DOCS -= 1))
```
Comment (primeroz) [inline resources/libs/vault.sh:32]: this was `N_DOCS` and changed to `_DOCS` but later is still `N_DOCS`

File: Dockerfile (hunk 1 above)
Comment (chrlwrd) [reply Dockerfile:89]: Added vim-tiny, plus bash alias to make it work

File: resources/libs/vault.sh (hunk 2 above)
Comment (chrlwrd) [reply resources/libs/vault.sh]: My bad, fixed

**Pull request merged**